torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -0,0 +1,194 @@
+#include <torch/csrc/jit/codegen/cuda/arith.h>
+#include <torch/csrc/jit/codegen/cuda/ir_iostream.h>
+#include <torch/csrc/jit/codegen/cuda/lower_utils.h>
+
+#include <torch/csrc/jit/codegen/cuda/lower_thread_predicate.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+
+const static std::unordered_map<ParallelType, int> pt_to_offset{
+    {ParallelType::BIDx, 0},
+    {ParallelType::BIDy, 1},
+    {ParallelType::BIDz, 2},
+    {ParallelType::TIDx, 3},
+    {ParallelType::TIDy, 4},
+    {ParallelType::TIDz, 5}};
+
+const static std::unordered_map<int, ParallelType> offset_to_pt{
+    {0, ParallelType::BIDx},
+    {1, ParallelType::BIDy},
+    {2, ParallelType::BIDz},
+    {3, ParallelType::TIDx},
+    {4, ParallelType::TIDy},
+    {5, ParallelType::TIDz}};
+
+static constexpr int num_p_type = 6;
+
+namespace {
+
+void flip_true(std::bitset<num_p_type>& bits, const ParallelType p_type) {
+  if (pt_to_offset.find(p_type) == pt_to_offset.end()) {
+    TORCH_INTERNAL_ASSERT(false, "Could not recognize parallel type.");
+  }
+  bits[pt_to_offset.at(p_type)] = true;
+}
+
+Val* threadPredicate(int i) {
+  if (offset_to_pt.find(i) == offset_to_pt.end()) {
+    TORCH_INTERNAL_ASSERT(
+        false,
+        "Invalid int for predicate computation, should be from [0-5], but recieved, ",
```

Comment:
hard-coding limits in strings :) an out of range or just "invalid" value would suffice IMO